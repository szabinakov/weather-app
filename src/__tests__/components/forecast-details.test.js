import React from 'react'
import { render } from "@testing-library/react"
import ForecastDetails from '../../components/forecast-details'

describe('Forecast details', () =>{
    const mockForecast = {
        date: 1596747600000,
        temperature:{
            max: 35,
            min: 12
        },
        wind: {
            speed: 5,
            direction: 's'
        },
        humidity:88
    }
    it('matches the snapshot', () => {
        const { asFragment } = render ( <ForecastDetails forecast={mockForecast}/>)
        expect(asFragment()).toMatchSnapshot()
    })
    it('renders the correct amount of props for ForecastDetails', () => {
        const { getByTestId } = render ( <ForecastDetails forecast={mockForecast}/>)

        expect(getByTestId('dateId')).toHaveClass('dateClass')
        expect(getByTestId('temperatureId')).toHaveClass('temperatureClass')
        expect(getByTestId('windId')).toHaveClass('windClass')
        expect(getByTestId('humidityId')).toHaveClass('humidityClass')
    })
})