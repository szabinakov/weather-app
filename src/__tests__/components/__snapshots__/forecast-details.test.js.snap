// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast details matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecastdetails"
  >
    <div
      class="dateClass"
      data-testid="dateId"
    >
      Thursday 6th Aug
    </div>
    <div
      class="temperatureClass"
      data-testid="temperatureId"
    >
      Max-Temperature:
      <br />
       35°
      <br />
      Min-Temperature:
      <br />
       12°
    </div>
    <div
      class="windClass"
      data-testid="windId"
    >
      Wind:
      <br />
       5 km/h s
    </div>
    <div
      class="humidityClass"
      data-testid="humidityId"
    >
      Humidity:
      <br />
       88 %
    </div>
  </div>
</DocumentFragment>
`;
